pkg/controllers/disruption/emptynodeconsolidation.go

```diff
@@ -79,14 +90,21 @@ func (c *EmptyNodeConsolidation) ComputeCommand(ctx context.Context, candidates
 	// We do this so that we can re-validate that the candidates that were computed before we made the decision are the same
 	candidatesToDelete := mapCandidates(cmd.candidates, validationCandidates)
 
+	postValidationMapping, err := buildDisruptionBudgets(ctx, c.cluster, c.clock, c.kubeClient)
```

Comment:
This validation delay here definitely scares me. What are the edge-cases here w.r.t. to re-building the disruption budgets mapping? Are there any risks that this might continually fail for some reason?

Reply:
What about the delay scares you? We're checking the budgets before and after the validation, which I think is the best we can do.